### ISSUE TITLE
Unable To Change Audio Music Track Volume

### ISSUE THREAD

**Issue by MrTertius** [OpenShot/openshot-qt#1576]
**Describe the bug**
There's a problem where if i try to change an Audio Music Track, the volume will not be changed

**System Details (please complete the following information):**
 - Operating System / Distro: Windows 10
 - OpenShot Version 2.4.1-90

**To Reproduce**
1. Go to the audio track on the timeline
2. Right Click on the audio track and select properties
3. Scroll down to Volume in the Properties Tab and change the value
4. See the error by playing the video

**Expected behavior**
The volume of the audio track would have changed

**Screenshots**

**Logs**
If you are experiencing a crash, please collect and attach logs of the problem.

**Additional context**
This is a error where in, most build versions, the Developer forgot to fix this problem

**Comment by N3WWN** [comment]
What values are you using for the `Volume` property?  

The volume is a percent, shown with 0% = `0.00` and 100% = `1.00`.  

If you want to reduce the volume to 50%, you would use `0.50` in the `Volume` property.

To boost the volume by 10%, you would use `1.10`.

**Comment by N3WWN** [comment]
@MrTertius - I just tested this with OpenShot-v2.4.1-115-ge7a99e9-59-697-x86_64.AppImage and OpenShot-v2.4.1-x86_64.AppImage.  The daily build will not let me change volume while the v2.4.1 release build will.  Thanks for finding this bug and reporting it!

@DylanC - I am thinking this is probably something that @jonoomph would want to be resolved before v2.4.2 is released.

Looking at the libopenshot changes since the 2.4.1 release, I think the issue lies here:

https://github.com/OpenShot/libopenshot/blob/1ec9e4426a94181187ac1f039971d7700911abb2/src/Frame.cpp#L865

...that line overwrites the value of `gainFactor`, which was previously set to the clip-specific value of `gainToApplyToSource` on line 862.  

I _think_ the fix may be to change `gainFactor = ...` to `gainFactor *= ...` at the start of L865 so that gainFactor is scaled appropriately instead of overwritten.  But, since I'm not familiar with that code, and I'm a little low on time today, I can't test that theory myself right now.

**Comment by DylanC** [comment]
@jonoomph - Might be worth looking at before v2.4.2 is released.

**Comment by DylanC** [comment]
Also found the same issue here, reported earlier #1533

**Comment by N3WWN** [comment]
Good catch, Sir @DylanC !

I *may* get chance to look at this today...

**Comment by N3WWN** [comment]
Okay... playing around with the code brought some things to light:

- With the changes made since v2.4.1 was released
  - Volume **changes** within a clip (preset fades in/out i.e. `Volume` -> `Start of Clip` -> `Fade In ...` or manual volume changes within the clip done with **multiple** keyframes) are behave properly because they are applied via `ApplyGainRamp()`
  - Fixed volume (`Volume` -> `Entire Clip` -> `Level ...` or setting a single volume keyframe at the start of the clip) settings are ignored due to https://github.com/OpenShot/libopenshot/blob/1ec9e4426a94181187ac1f039971d7700911abb2/src/Frame.cpp#L865 because fixed volumes are applied to the clip via `AddAudio()` and that line overwrites the `gainToApplyToSource` parameter of the function.
- `current_max_volume` doesn't appear to be computed correctly, but I'm not familiar with what it is supposed to do, so I'm not going to touch it.  I have added a `TODO`  note in the source, though.

I will be submitting a PR to fix the fixed volume issues introduced since v2.4.1 was published with hopes that it gets into v2.4.2 before it is released.
